### PR TITLE
Recursive algorithm for finding punishment number

### PR DESCRIPTION
**Pull request by aeruru**

### CODE REVIEW

File: 2698. Find the Punishment Number of an Integer/solution.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+class Solution:
+    # Given a positive integer n, return the punishment number of n which
+    # is determined by adding up the squares of any integer i that is 
+    # between 1 and n (inclusive) and the following complicated condition:
+    # The square of i can be partitioned into contiguous substrings so that
+    # the sum of the integers values of the substrings is equal to i.
+    def punishmentNumber(self, n: int) -> int:
+
```
Comment (winry-fruitiwi) [inline 2698. Find the Punishment Number of an Integer/solution.py:8]: nit: extra whitespace

File: 2698. Find the Punishment Number of an Integer/solution.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            # included in punishment number.
+            if Solution.canBePartitionedRecursive(i):
+                # if criteria is met, add the square of i to punishment number
+                punishmentNum += i * i
+
+        return punishmentNum
+
+    # Given an int i, return whether the string representation of i squared can
+    # be partitioned into substrings such that the sum of the integer value
+    # of the substrings equals i. This is an interative algorithm.
```
Comment (winry-fruitiwi) [inline 2698. Find the Punishment Number of an Integer/solution.py:22]: nit: please proofread typos

File: 2698. Find the Punishment Number of an Integer/solution.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+            firstDigitsAsInt = int(firstDigits)
+            remainingDigits = s[x:]
+            # recursively call helper with the remaining digits as a string and
+            # with the target being the current target minus the integer value
+            # of the first digits
+            if Solution.canBePartitionedRecursiveHelper(s=remainingDigits, target=target - firstDigitsAsInt):
+                return True
+
+        # no partition that met target could be found so return False.
+        return False
```
Comment (winry-fruitiwi) [inline 2698. Find the Punishment Number of an Integer/solution.py:71]: nit: needs whitespace

**Review by winry-fruitiwi — COMMENTED** [review]
Just a bunch of nits

File: 2698. Find the Punishment Number of an Integer/solution.py (hunk 1 above)
Comment (aeruru) [reply 2698. Find the Punishment Number of an Integer/solution.py:8]: @winry-fruitiwi - Where's the extra white space?

File: 2698. Find the Punishment Number of an Integer/solution.py (hunk 2 above)
Comment (aeruru) [reply 2698. Find the Punishment Number of an Integer/solution.py:22]: This comment is really annoying. Just point out the typos -_-

File: 2698. Find the Punishment Number of an Integer/solution.py (hunk 3 above)
Comment (aeruru) [reply 2698. Find the Punishment Number of an Integer/solution.py:71]: What needs white space?

**Pull request merged**